### PR TITLE
feat: handle rune

### PR DESCRIPTION
**Pull request by nythrox**
#790 

possible changes: 
- change name of `RunMatch` (RunHandle is already taken, maybe refactor)
- handleError function

### CODE REVIEW

File: rune/ValueRune.ts [hunk 1]
```diff
@@ -53,6 +53,18 @@ export class ValueRune<out T, out U = never> extends Rune<T, U> {
...
   }
 
+  handle<T2 extends T, T3, U2>(
+    guard: Guard<T, T2>,
+    alt: (rune: ValueRune<T2, never>) => Rune<T3, U2>,
+  ): ValueRune<Exclude<T, T2> | T3, U | U2> {
+    return ValueRune.new(
+      RunMatch,
+      this,
+      (x: T): x is T2 => checkGuard(x, guard),
```
Comment (tjjfvi) [inline rune/ValueRune.ts:63]: This should pass `guard` directly like in `RunUnhandle`

File: rune/ValueRune.ts [hunk 2]
```diff
@@ -53,6 +53,18 @@ export class ValueRune<out T, out U = never> extends Rune<T, U> {
...
+      ValueRune.new(RunChain, this, alt(this as never)),
```
Comment (tjjfvi) [inline rune/ValueRune.ts:64]:
```suggestion
      alt(this as never),
```

File: rune/ValueRune.ts [hunk 3]
```diff
@@ -151,6 +163,28 @@ class RunMap<T1, U, T2> extends Run<T2, U> {
   }
 }
 
+class RunMatch<T, T2 extends T, T3, U, U2> extends Run<Exclude<T, T2> | T3, U | U2> {
```
Comment (tjjfvi) [inline rune/ValueRune.ts:166]: Yeah, this should be called `RunHandle`, and the existing one should be renamed to `RunRehandle`

File: rune/ValueRune.ts (hunk 1 above)
Comment (tjjfvi) [reply rune/ValueRune.ts:63]: This wasn't resolved

Comment (nythrox) [reply rune/ValueRune.ts:63]: oops my bad, I updated the Run function for this and forgot to update it here

**Review by tjjfvi — APPROVED** [review]

**Pull request merged**